trieste/acquisition/rule.py

```diff
@@ -2005,18 +2118,26 @@ def __init__(
             )
 
         self._global_search_space = TaggedProductSearchSpace(
-            [region.global_search_space for region in regions]
+            [region.global_search_space for region in regions], tags
         )
 
-        # Find the most general dtype of all the regions.
-        dtypes = [np.array(region.location).dtype for region in regions]
-        most_general_dtype = np.result_type(*dtypes)
-        self.dtype = most_general_dtype
-
         TaggedProductSearchSpace.__init__(self, regions)
         # When UpdatableTrustRegion sets the region_index, it will also set the region_index for
         # each region.
-        UpdatableTrustRegion.__init__(self, region_index)
+        # Setting of input active dims is not supported for product regions. All input dims
+        # are always active.
+        UpdatableTrustRegion.__init__(self, region_index, input_active_dims=None)
+
+        # Set active dimensions for each sub-region.
+        dim_ix = 0
+        for region, dims in zip(regions, self.subspace_dimension):
+            # Check the region's input active dims are not already set.
+            assert region.input_active_dims is None, (
+                f"input_active_dims ({region.input_active_dims}) should not be set for sub-regions "
+                f" ({region}) of a product region"
+            )
+            region.input_active_dims = slice(dim_ix, dim_ix + dims)
+            dim_ix += dims
```

Comment:
shouldn't this be informed by the global search space, which dimensions exactly are discrete there?

Reply:
It is already informed by the global search space, effectively. `self.subspace_dimension` returns the list of dimensions of all the subspaces/regions. The same subspace/regions are used a few lines above to create `self._global_search_space`.

Also, here we are creating the active dims for all type search spaces, not just discrete.